src/SourceLocator/Type/AbstractSourceLocator.php

```diff
@@ -54,15 +55,15 @@ public function locateIdentifier(Reflector $reflector, Identifier $identifier):
      *
      * @throws ParseToAstFailure
      */
-    final public function locateIdentifiersByType(Reflector $reflector, IdentifierType $identifierType): array
+    final public function locateIdentifiersByType(Reflector $reflector, IdentifierType $identifierType): Generator
```

Comment:
I'd suggest declaring `iterable` here too: `Generator` is a very concrete type.

We can `yield from` any `iterable` anyway, and we don't need to force every implementation into a `Generator`.

See https://3v4l.org/j4S0l